### PR TITLE
feat: add projects API functionality

### PR DESCRIPTION
**Pull request by vorobeyme**

### CODE REVIEW

File: crowdin/model/projects.go [hunk 1]
```diff
@@ -0,0 +1,525 @@
...
+	// segmentContext - searching by context.
+	// auto - context search for key-value formats and segment search for others.
+	// prevAndNextSegment - search by previos and next segment.
+	TMContextType  string                `json:"tmContextType,omitempty"`
+	TMPreTranslate ProjectTMPreTranslate `json:"tmPreTranslate,omitempty"`
+	MTPreTranslate ProjectMTPreTranslate `json:"mtPreTranslate,omitempty"`
+	// Context and max.length added in Crowdin will be visible in the downloaded files.
+	SaveMetaInfoInSource bool `json:"saveMetaInfoInSource,omitempty"`
+	// Defines the project type. To create a file-based project, type should be set to 0.
+	// Enum: 0, 1. Default: 0.
```
Comment (andrii-bodnar) [inline crowdin/model/projects.go:225]:
```suggestion
	// Defines the project type. Use 0 for a file-based project and 1 for a string-based project.
	// Enum: 0, 1. Default: 0.
```

File: crowdin/projects.go [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+}
+
+// Delete deletes a project by its identifier.
+//
+// https://developer.crowdin.com/api/v2/#operation/api.projects.delete
+func (s *ProjectsService) Delete(ctx context.Context, id int64) (*Response, error) {
+	return s.client.Delete(ctx, fmt.Sprintf("/api/v2/projects/%d", id))
+}
+
+// https://developer.crowdin.com/api/v2/#operation/api.projects.file-format-settings.custom-segmentations.get
```
Comment (andrii-bodnar) [inline crowdin/projects.go:88]: description is missing

File: crowdin/projects.go [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+	path := fmt.Sprintf("/api/v2/projects/%d/file-format-settings/%d/custom-segmentations", projectID, settingsID)
+	res := struct {
+		Data *model.DownloadCustomSegmentations `json:"data"`
+	}{}
+	resp, err := s.client.Get(ctx, path, nil, res)
+
+	return res.Data, resp, err
+}
+
+// https://developer.crowdin.com/api/v2/#operation/api.projects.file-format-settings.custom-segmentations.delete
```
Comment (andrii-bodnar) [inline crowdin/projects.go:100]: description is missing

File: crowdin/model/projects.go (hunk 1 above)
Comment (vorobeyme) [reply crowdin/model/projects.go]: Thank you, it looks better

File: crowdin/projects.go (hunk 2 above)
Comment (vorobeyme) [reply crowdin/projects.go:91]: thanks, I've updated the description

File: crowdin/projects.go (hunk 3 above)
Comment (vorobeyme) [reply crowdin/projects.go:106]: done

**Review by andrii-bodnar — APPROVED** [review]

**Pull request merged**